src/components/Post.js

```diff
@@ -14,24 +14,40 @@ class Post extends Component {
     this.setState({ isEditModeOn: !this.state.isEditModeOn });
   };
   render() {
-    const { title, body, tags } = this.props.post.entry;
+    const { title, body, tags, userId } = this.props.post.entry;
+
     const postUuid = this.props.post.entry.uuid;
+
     return (
       <div className="blogSeparator">
         {this.state.isEditModeOn ? (
-          <BlogForm onUpdateClick={this.changeState} entry={this.props.post.entry} entryUuid={postUuid} />
+          <BlogForm
+            onUpdateClick={this.changeState}
+            entry={this.props.post.entry}
+            entryUuid={postUuid}
+            userId={userId}
+          />
         ) : (
           <div>
             <div className="postTitle">
               <span>
                 <h2>{title}</h2>
               </span>
               <span>
-                <EditEntry onEditClick={this.changeState} postUuid={postUuid} />
-                <DeleteEntry onEditClick={this.changeState} postUuid={postUuid} />
+                {this.props.auth.uid === userId && (
+                  <span>
+                    <EditEntry onEditClick={this.changeState} postUuid={postUuid} />
+                    <DeleteEntry onEditClick={this.changeState} postUuid={postUuid} />
+                  </span>
+                )}
               </span>
             </div>
             <div className="postContent">
+              {this.props.users.map(user => {
```

Comment:
This is not really what you want to do here. You actually want to **find** the user with that ID and then add the explanatory text. Make the code obvious.